CONTRIBUTING.rst

```diff
@@ -1,3 +1,42 @@
+Development Quickstart
```

Comment:
We already have something like this [here](http://docs.hylang.org/en/latest/hacking.html). Anything here should probably go over in `docs/hacking.rst`.